framework/src/constraints/MortarConstraintBase.C

```diff
@@ -90,7 +90,7 @@ MortarConstraintBase::MortarConstraintBase(const InputParameters & parameters)
     _normals(_assembly.normals()),
     _tangents(_assembly.tangents()),
     _JxW_msm(_assembly.jxWMortar()),
-    _coord(_assembly.coordTransformation()),
+    _coord(_assembly.mortarCoordTransformation()),
```

Comment:
Maybe this?

Reply:
No. If you don't have this then `_coord` will actually be the wrong size because it's using a volumetric quadrature rule. But with respect to Jacobians...`_coord` has always been stored as a `MooseArray<Real>` so there has never been derivative information there.